src/core/cgroup.c

```diff
@@ -181,6 +182,270 @@ void cgroup_context_init(CGroupContext *c) {
         };
 }
 
+int cgroup_context_copy(const CGroupContext *src, CGroupContext *dst) {
+        struct in_addr_prefix *i;
+        char *iface;
+        int r;
+
+        assert(src);
+        assert(dst);
+
+        dst->cpu_accounting = src->cpu_accounting;
+        dst->io_accounting = src->io_accounting;
+        dst->blockio_accounting = src->blockio_accounting;
+        dst->memory_accounting = src->memory_accounting;
+        dst->tasks_accounting = src->tasks_accounting;
+        dst->ip_accounting = src->ip_accounting;
+
+        dst->memory_oom_group = dst->memory_oom_group;
+
+        dst->delegate = src->delegate;
+        dst->delegate_controllers = src->delegate_controllers;
+        dst->disable_controllers = src->disable_controllers;
+        dst->delegate_subgroup = src->delegate_subgroup;
+
+        dst->cpu_weight = src->cpu_weight;
+        dst->startup_cpu_weight = src->startup_cpu_weight;
+        dst->cpu_quota_per_sec_usec = src->cpu_quota_per_sec_usec;
+        dst->cpu_quota_period_usec = src->cpu_quota_period_usec;
+
+        dst->cpuset_cpus = src->cpuset_cpus;
+        dst->startup_cpuset_cpus = src->startup_cpuset_cpus;
+        dst->cpuset_mems = src->cpuset_mems;
+        dst->startup_cpuset_mems = src->startup_cpuset_mems;
+
+        dst->io_weight = src->io_weight;
+        dst->startup_io_weight = src->startup_io_weight;
+
+        LIST_FOREACH(device_weights, w, src->io_device_weights) {
+                _cleanup_free_ CGroupIODeviceWeight *n = NULL;
+
+                n = new0(CGroupIODeviceWeight, 1);
+                if (!n)
+                        return -ENOMEM;
+
+                n->path = strdup(w->path);
+                if (!n->path)
+                        return -ENOMEM;
+                n->weight = w->weight;
+
+                LIST_PREPEND(device_weights, dst->io_device_weights, TAKE_PTR(n));
```

Comment:
Hmm, this reverses the list. That seems a bit ugly. Use LIST_FOREACH_BACKWARDS instead?

Reply:
I kept for now because this might get deleted anyway if @poettering insists.

Reply:
If we're going to merge this, please reverse it.